x/derivatives/keeper/perpetual_futures.go

```diff
@@ -519,10 +518,14 @@ func (k Keeper) ConvertBaseAmountToQuoteAmount(ctx sdk.Context, market types.Mar
 
 func (k Keeper) GetPerpetualFuturesPositionSizeInMetrics(ctx sdk.Context, market types.Market, pType types.PositionType) sdk.Dec {
 	perpFuturesLongPositionNum := k.GetPerpetualFuturesGrossPositionOfMarket(ctx, market, pType)
-	baseDenomPrice, err := k.GetPrice(ctx, market.BaseDenom, market.QuoteDenom)
+	currentBaseUsdRate, currentQuoteUsdRate, err := k.GetPairUsdPriceFromMarket(ctx, market)
 	if err != nil {
 		return sdk.ZeroDec()
 	}
-
-	return sdk.NewDecFromInt(perpFuturesLongPositionNum.PositionSizeInDenomExponent).Mul(baseDenomPrice.Price)
+	baseDenomPrice := currentBaseUsdRate.Quo(currentQuoteUsdRate)
```

Comment:
@taiki1frsh can you check is this modification valid?

Reply:
@KimuraYu45z 
Although I don't know why this part needs to be changed, I think this code should also works fine.